In [1]:
from openpyxl import load_workbook
import os
import pandas as pd

In [2]:
files = {'H:/FYP DATASETS/FYP DATASETS/Crisis/annotations/mexico_earthquake_final_data.xlsx',
         'H:/FYP DATASETS/FYP DATASETS/Crisis/annotations/iraq_iran_earthquake_final_data.xlsx',
         'H:/FYP DATASETS/FYP DATASETS/Crisis/annotations/hurricane_irma_final_data.xlsx',
         'H:/FYP DATASETS/FYP DATASETS/Crisis/annotations/srilanka_floods_final_data.xlsx',
         'H:/FYP DATASETS/FYP DATASETS/Crisis/annotations/hurricane_maria_final_data.xlsx',
         'H:/FYP DATASETS/FYP DATASETS/Crisis/annotations/hurricane_harvey_final_data.xlsx',
         'H:/FYP DATASETS/FYP DATASETS/Crisis/annotations/california_wildfires_final_data.xlsx'}
sheet_name = 'Sheet 1'

severe_damage = 'filtered/crisis/severe_damage/'
mild_damage = 'filtered/crisis/mild_damage/'
little_damage = 'filtered/crisis/little_damage/'
dont_know = 'filtered/crisis/dont_know/'

In [3]:
df = pd.DataFrame(columns=['text', 'text_info', 'image_path', 'damage'])
print(df.head)

<bound method NDFrame.head of Empty DataFrame
Columns: [text, text_info, image_path, damage]
Index: []>


In [4]:
image_path = 'O'
tweet_text = 'M'
image_damage = 'K'
image_type = 'I'
text_info = 'C'

In [5]:
damages_scale = ['severe_damage', 'mild_damage', 'little_or_no_damage']
text_class = ['informative', 'not_informative']
for file in files:
    wb = load_workbook(file, data_only=True)
    sh = wb[sheet_name]
    row_count = sh.max_row
    for i in range(3,row_count):
        if sh[text_info + str(i)].value in text_class:
            if sh[image_type + str(i)].value == 'infrastructure_and_utility_damage':
                if sh[image_damage + str(i)].value in damages_scale:
                    df = df.append({'text' : sh[tweet_text + str(i)].value,
                                        'text_info' : sh[text_info + str(i)].value,
                                        'image_path' : sh[image_path + str(i)].value,
                                        'damage': sh[image_damage + str(i)].value}, ignore_index=True)
            else:
                df = df.append({'text' : sh[tweet_text + str(i)].value,
                                    'text_info' : sh[text_info + str(i)].value,
                                    'image_path' : sh[image_path + str(i)].value,
                                    'damage': 'irrelevant'}, ignore_index=True)


In [6]:
df = df.sample(frac=1).reset_index(drop=True)
informative_count = 0
scale_count = 0
# print(df.head)
for index, row in df.iterrows():
    if row['text_info']!= 'informative':
        informative_count += 1
    if row['damage'] in damages_scale:
        scale_count += 1
    
print('informative text  count ' + str(informative_count))
print('damage scale count ' + str(scale_count))
print('\n\n')

print(df['damage'].value_counts())


informative text  count 5222
damage scale count 3533



irrelevant             14480
severe_damage           2216
mild_damage              842
little_or_no_damage      475
Name: damage, dtype: int64


In [7]:
df.to_pickle("./dataset.pkl")
print("dataframe saved")

dataframe saved


In [8]:
print(df.iloc[1])

text          Ragin’ Cajun athletes reaching out and helpi...
text_info                                           informative
image_path    data_image/hurricane_harvey/15_9_2017/90855601...
damage                                               irrelevant
Name: 1, dtype: object


In [9]:
# df_sorted = df.sort_values(by=['text_info'], ascending = False)

In [10]:
print(df.damage.unique())
print(df.text_info.unique())

['irrelevant' 'severe_damage' 'mild_damage' 'little_or_no_damage']
['informative' 'not_informative']


informative text  count 5229
damage scale count 3533
text          18020
text_info     18020
image_path    18020
damage        18020
dtype: int64